redcap/project.py

```diff
@@ -149,6 +157,423 @@ def _call_api(self, payload, typpe, **kwargs):
         rcr = RCRequest(self.url, payload, typpe)
         return rcr.execute(**request_kwargs)
 
+    def __toggle_dataframe(self,format):
+        ret_format = format
+        if format == 'df':
+            try:
```

Comment:
This  can be hoisted to the top of the file. If pandas doesn't import, catch and set `read_csv` to None. Then in export methods we can `if not read_csv and format == 'df': blow up`.